cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -317,22 +317,22 @@ subroutine evp (dt)
       ! convert fields from T to U grid
       !-----------------------------------------------------------------
 
-      call grid_average_X2Y('F', tmass    , 'T'          , umass   , 'U')
-      call grid_average_X2Y('F', aice_init, 'T'          , aiU     , 'U')
+      call grid_average_X2Y('S', tmass    , 'T'          , umass   , 'U')
+      call grid_average_X2Y('S', aice_init, 'T'          , aiU     , 'U')
```

Comment:
The change here for tmass is right: to get the correct mass at the U point, we need to compute 
`sum(rhoi*vice*aice*cell area)/sum(aice*cell area)`
over the T cells [ignoring snow for now], since vice has units of volume per unit area.

But I do not think this is correct for aice_init.  Since ice concentration is assumed to be uniformly distributed over each grid cell, the concentration at U should be
`sum(aice_init*cell area)/sum(cell area)`
but I think the calculation here is 
`sum(aice_init*aice*cell area)/sum(aice*cell area)`.

Reply:
Just to clarify, The "S" mapping is masked area weighted while the "A" mapping is just area weighted.  When you use "aice" above, is that the ice fraction?  There are no ice fraction weighted averages built into CICE.  If we want to do that, we can use user defined averaging weights.  Again,

```
"S":  sum(var*Tmask*Tarea)/sum(Tmask*Tarea)
"A":  sum(var*Tarea)/sum(Tarea)
"F":  sum(var*Tarea_var)/Tarea_dst
```

Where Tarea and Tmask are the static grid cell areas and masks computed at initialization.  If we do want to have fraction weighted mapping, we can do that, but it's not the standard "S", "A", or "F".  To do that we'd do

```
mask_s(:,:,;) = 1
wght_s(:,:,:) = Tarea(:,:,:) * aice(:,:,:)
call grid_average_X2Y('A', tmass, 'T', wght_s, mask_s, umass, 'U')
```

and that would do

`umass = sum(tmass*Tarea*aice)/sum(Tarea*aice)
`

At this point, we NEVER weigh the grid averaging by the ice fraction anywhere.  Is that something we should be doing?

Reply:
Sorry, I misunderstood "area weighted" in the doc to mean ice-area-weighted, but it clearly says grid cell area.  My mistake.  Let me think about this some more - what I wrote above is not quite correct.

Reply:
Correction to my comment above: 

To get the correct mass at the U point, we need to compute 
`sum(rhoi*vice*Tarea)/sum(Tarea)` 
over the T cells [ignoring snow for now], since vice has units of volume per unit area and already incorporates aice (i.e. vice = hi * aice).  So the tmass line is correct.

Since ice concentration is assumed to be uniformly distributed over each grid cell, the initial concentration at U should be 
`sum(aice_init*Tarea)/sum(Tarea)` 
and that is what's being done in the S mapping. 

So I agree with both of these modifications.  Sorry for the confusion!

Reply:
The "S" mapping includes the mask, the "A" mapping does not.  I think we need to change "S" to "A" for umass and aiU if we want an unmasked weighted average as written above.  This only matters for gridcells near land.  So should the average be an average of only ocean cells ("S" mapping) or an average of all ocean/land cells ("A" mapping)?

One other thing that I just want reiterate.  Maybe there is a place for fraction weighted mapping in CICE.  We do that in the coupler when we map ice coupling fields to other grids.  We fraction area weight the mapping so a gridcell with fraction=0.9 has much greater weight than a gridcell with fraction=0.2 when interpolating from ice to atmosphere for example.  You can also think of it as a fraction weighted merge.  How we do that depends whether it's a flux (which has to be strictly conserved and has a particular mapping requirement) or a state (which could be done many ways). Maybe we need to do the same in CICE in some places for some fields??  I guess it partly depends whether we can define an ice fraction on the grid we're mapping from.  We'd have to think about it a bit.

Reply:
I think S is correct in this case, with the mask included, although it would be good for @JFLemieux73 @dabail10 to confirm that it's consistent with other assumptions in the C-grid approach.  On a B-grid, the velocity being computed would be zero if any of the T cells is land (i.e. the point is moot).

Reply:
Great, thanks for the clarification.

Reply:
I also wondered whether ice area weighting could/should be included as part of the averaging infrastructure.  For coupling, the actual averaging happens in the coupler with other components' data, so we can't control the entire average anyhow. Once the model starts being coupled with different grids, we should check whether we are calculating the coupling fields appropriately, but I think you've already thought through all of that.  In the ice model, we "aggregate" across ice thickness categories, but that's only within a particular cell, not for shifting things around horizontally (there are vertical grid remappings). The tracer loading/unloading also involves a lot of multiplying/dividing by "weights" (including ice area and other tracers).  Neither of these cases involve grid shifts.  So: I don't think we should worry about it unless we come across a particular case in which we need it.